akka-projection-r2dbc-integration/src/test/scala/akka/projection/r2dbc/EventSourcedEndToEndSpec.scala

```diff
@@ -166,24 +239,81 @@ class EventSourcedEndToEndSpec
     result.futureValue shouldBe 1
   }
 
+  private def startExactlyOnceProjections(startParams: StartParams): Vector[ActorRef[ProjectionBehavior.Command]] = {
+    startProjections(
+      startParams,
+      (projectionId, sourceProvider) =>
+        R2dbcProjection
+          .exactlyOnce(
+            projectionId,
+            Some(projectionSettings),
+            sourceProvider = sourceProvider,
+            handler =
+              () => new TestHandler(projectionId, startParams.processedProbe.ref, processedEvents(projectionId))))
+  }
+
+  private def startAtLeastOnceProjections(startParams: StartParams): Vector[ActorRef[ProjectionBehavior.Command]] = {
+    startProjections(
+      startParams,
+      (projectionId, sourceProvider) =>
+        R2dbcProjection
+          .atLeastOnce(
```

Comment:
we also have some async variants, worth adding all?
atLeastOnceAsync
groupedWithinAsync

Reply:
Wasn't sure, but may as well cover all the variants now.